### DISCUSSION TITLE
delete labels / Weights from a trained model

### DISCUSSION THREAD

**Discussion by BaAL94** [online-ml/river#1024] [Q&A] [answered]
Hey River community,

lets say im using my model in the production for a multiclass classification problem and after a while it turned out that the model was somehow fed with wrong informations eg. wrong labels. so my question is, what should I do in this case? should I retrain the model from the scratch? is there a way to delete/extract the wrong labels from my model?

Thank you!

**Comment by gbolmier** [comment, marked as answer]
Hi @BaAL94, unfortunately there is no way to "cancel" specific learning steps a model performed. The best is to prevent such issues to happen if you can. Otherwise, one way could be to make regular snapshots of your model and revert to one of them when it happens.

**Reply by MaxHalford** [reply]
Good answer. Also note that an online model implicitly forgets about past observations. So the impact of erroneous sample will fade with time.

**Reply by Robert-Schimanek** [reply]
@MaxHalford  Is implicit forgetting somewhere documented?

**Reply by MaxHalford** [reply]
No, not really @Robert-Schimanek.  What kind of explanation would be looking for?
